File: packages/java/examples/04_multi_agent/ReporterAgent.java
import com.a2alite.Agent;
import com.a2alite.SkillConfig;
import java.util.*;

/**
 * Reporter Agent - Part of multi-agent example.
 *
 * Run: ./gradlew run -PmainClass=multi_agent.ReporterAgent
 */
class ReporterAgent {
    public static void main(String[] args) {
        var agent = Agent.builder().name("ReporterAgent").description("Generates reports").build();

        agent.skill("generate_summary", SkillConfig.of("Generate summary report", List.of("reporting")), params -> {
            @SuppressWarnings("unchecked")
            Map<String, Object> data = (Map<String, Object>) params.get("data");
            String format = (String) params.getOrDefault("format", "text");
            StringBuilder sb = new StringBuilder();
            if ("markdown".equals(format)) {
                sb.append("# Summary Report\n\n");
                data.forEach((k, v) -> sb.append("- **").append(k).append("**: ").append(v).append("\n"));
            } else {
                sb.append("Summary Report\n").append("=".repeat(40)).append("\n");
                data.forEach((k, v) -> sb.append(k).append(": ").append(v).append("\n"));
            }
            return Map.of("report", sb.toString(), "format", format);
        });

        agent.skill("format_table", SkillConfig.of("Format as table", List.of("reporting")), params -> {
            @SuppressWarnings("unchecked")
            List<String> headers = (List<String>) params.get("headers");
            @SuppressWarnings("unchecked")
            List<List<Object>> rows = (List<List<Object>>) params.get("rows");
            int[] widths = new int[headers.size()];
            for (int i = 0; i < headers.size(); i++) {
                final int idx = i;
                int maxData = rows.stream().mapToInt(r -> String.valueOf(r.get(idx)).length()).max().orElse(0);
                widths[i] = Math.max(headers.get(i).length(), maxData);
            }
            StringBuilder sb = new StringBuilder();
            String sep = "+" + Arrays.stream(widths).mapToObj(w -> "-".repeat(w + 2)).reduce((a,b)->a+"+"+b).orElse("") + "+\n";
            sb.append(sep);
            for (int i = 0; i < headers.size(); i++) sb.append("| ").append(String.format("%-" + widths[i] + "s", headers.get(i))).append(" ");
            sb.append("|\n").append(sep);
            for (var row : rows) {
                for (int i = 0; i < row.size(); i++) sb.append("| ").append(String.format("%-" + widths[i] + "s", row.get(i))).append(" ");
                sb.append("|\n");
            }
            sb.append(sep);
            return Map.of("table", sb.toString());
        });

        agent.run(8789);
    }
}

File: packages/typescript/examples/04_multi_agent/reporter_agent.ts
/**
 * Reporter Agent - Part of multi-agent example.
 *
 * Run: npx ts-node examples/04_multi_agent/reporter_agent.ts
 */
import { Agent } from '../../src';

const agent = new Agent({ name: 'ReporterAgent', description: 'Generates reports' });

agent.skill('generate_summary', { description: 'Generate summary report', tags: ['reporting'] }, async ({ data, format = 'text' }: { data: Record<string, unknown>; format?: string }) => {
  const lines: string[] = [];
  if (format === 'markdown') {
    lines.push('# Summary Report\n');
    for (const [key, value] of Object.entries(data)) {
      lines.push('- **' + key + '**: ' + JSON.stringify(value));
    }
  } else {
    lines.push('Summary Report');
    lines.push('='.repeat(40));
    for (const [key, value] of Object.entries(data)) {
      lines.push(key + ': ' + JSON.stringify(value));
    }
  }
  return { report: lines.join('\n'), format };
});

agent.skill('format_table', { description: 'Format as table', tags: ['reporting'] }, async ({ headers, rows }: { headers: string[]; rows: unknown[][] }) => {
  const widths = headers.map((h, i) => Math.max(h.length, ...rows.map(r => String(r[i] ?? '').length)));
  const sep = '+' + widths.map(w => '-'.repeat(w + 2)).join('+') + '+';
  const headerRow = '| ' + headers.map((h, i) => h.padEnd(widths[i])).join(' | ') + ' |';
  const dataRows = rows.map(row => '| ' + row.map((cell, i) => String(cell ?? '').padEnd(widths[i])).join(' | ') + ' |');
  return { table: [sep, headerRow, sep, ...dataRows, sep].join('\n') };
});

agent.run({ port: 8789 });
